src/bp/core/services/cms.ts

```diff
@@ -578,15 +578,49 @@ export class CMSService implements IDisposeOnExit {
 
   // This methods finds the translated property and returns the original properties
   private getOriginalProps(formData: object, contentType: ContentType, lang: string, defaultLang?: string) {
-    const originalProps = Object.keys(_.get(contentType, 'jsonSchema.properties'))
+    const schema = <FormDefinition>contentType.newSchema
+    const fields = [...schema.fields, ...schema.advancedSettings]
+    const result = _.cloneDeep(formData)
```

Comment:
It's necessary to deep clone otherwise the changes done on form data are still present on the next messages (is it cached somewhere?).